### PR TITLE
Add functions lesson

### PR DESCRIPTION
**Pull request by Oshawk**
Added a functions lesson and test.

### CODE REVIEW

**Comment by jackey8616** [comment]
@Oshawk 
I'm so so sorry that the test code is not designed will..
I saw your code and I run the test locally, and here is the error report:
```python
        req_data = json.dumps({
            'field_1': 'num1',
            'field_2': 'num2',
            'field_3': 'return'
        })
        res = await test_cli.post('/stage/one/functions', data=req_data)
        assert res.status == 200
        res_data = await res.json()
>       assert res_data['data']['result'] is True, '\n{0}'.format(json.dumps(res_data, indent=4))
E       AssertionError: 
E         {
E             "success": true,
E             "data": {
E                 "result": false,
E                 "stdout": [],
E                 "stderr": [
E                     "  File \"/home/clode/Program/YunOSC/PyFun-Backend/tmp/3ad931df7a64d52bef81f05bf7430bec.py\", line 2\n",                                                                                    
E                     "    return_ num1 + num2\n",
E                     "               ^\n",
E                     "SyntaxError: invalid syntax\n"
E                 ]
E             }
E         }
E       assert False is True
```

File: stage/one/s8_functions.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        'As you can see we start the function with the "def" keyword followed by the function name. ' +
+        'Next comes a set of brackets with the arguments for the function followed by a colon and the code.',
+        'The "return" keyword says what we want to give back to the user running the function.',
+        'To call a function, state the function name followed by brackets containing what you wish to pass in. ' +
+        'E.g. "is_odd(23)" which returns "True".',
+        'Now try to fill in the blanks below!!!'
+    ],
+    'code': [
+        'def add(_____, _____):',
+        '     ______ num1 + num2',
```
Comment (jackey8616) [inline stage/one/s8_functions.py]: Here is six of `_`, should be five.

**Comment by jackey8616** [comment]
@Oshawk 
s8 is already merged into master at PR #15 
You need to change to bigger index such as s9.

**Comment by Oshawk** [comment]
Apologies, hope it works now.

**Comment by jackey8616** [comment]
@Oshawk 
The test code have massively changed.
Please check our CONTRIBUTING.md again.
Sorry for that. 😢 
Also, do forgot re-index file name.

**Comment by Oshawk** [comment]
Think that is everything fixed now.

**Comment by jackey8616** [comment]
@Oshawk 
It's very fascinated! A Big thank you! 😃

**Pull request merged**